clientv3/watch.go

```diff
@@ -61,6 +61,11 @@ func (wr *WatchResponse) Err() error {
 	return nil
 }
 
+// IsProgressNotify returns true if the WatchResponse is progress notification.
+func (wr *WatchResponse) IsProgressNotify() bool {
+	return len(wr.Events) == 0 && !wr.Canceled
```

Comment:
Let me elaborate the sequence:

1. client or server is about the cancel the channel
2. client got the event of `Canceled == true`
3. client saw the channel is closed

Am I right? 

If so, I would add some docs to it from user's perspective. What does it mean to user though?